garden_ai/pipelines.py

```diff
@@ -88,324 +73,37 @@ class Paper(BaseModel):
-
-        To run a Pipeline on a remote endpoint, see ``RegisteredPipeline``.
-
-        Args:
-            *args : Any
-                Input data passed through the first step in the pipeline
-            **kwargs : Any
-                Additional keyword arguments passed directly to the first step in the pipeline.
-        Returns:
-            Results from the pipeline's composed steps called with the given input data.
-
-        Raises:
-            Exception:
-                Any exception raised over the course of executing the pipeline's composed steps.
-        """
-        has_models = len(self.model_full_names) > 0
-        if has_models:
-            if not garden_client:
-                raise ValueError("Missing required kwarg 'garden_client'")
-
-        return self._composed_steps(*args, **kwargs)
-
-    def __post_init_post_parse__(self):
-        # Finish initializing the pipeline after validators have run.
-        # - Build a single composite function from this pipeline's steps
-        # - Update metadata like signature, authors w/r/t underlying steps
-        # - Infer conda and pip dependencies from steps and requirements file
-        self._composed_steps = reduce(safe_compose, reversed(self.steps))
-        self.__signature__ = signature(self._composed_steps)
-        self._sync_author_metadata()
-        self._collect_requirements()
-        return
-
-    def _sync_author_metadata(self):
-        known_authors = set(self.authors)
-        known_contributors = set(self.contributors)
-        for step in self.steps:
-            new_contributors = set(step.authors) | set(step.contributors)
-            known_contributors |= new_contributors - known_authors
-        self.contributors = list(known_contributors)
-        return
-
-    def json(self) -> JSON:
-        """Helper: serialize pipeline metadata to JSON string."""
-        self._sync_author_metadata()
-        return json.dumps(self, default=garden_json_encoder)
-
-    def dict(self) -> Dict[str, Any]:
-        """Helper: serialize pipeline metadata to dictionary."""
-        d = {}
-        for key in self.__dataclass_fields__:
-            val = getattr(self, key)
-            if key == "steps":
-                val = [s.dict() for s in val]
-            d[key] = val
-        return d
-
-
 class RegisteredPipeline(BaseModel):
-    """Metadata of a completed and registered `Pipeline` object. Can be added to a Garden and execute on a remote Globus Compute endpoint.
-
-    Unlike `Pipelines`, `RegisteredPipelines` can be described completely by JSON (since they don't need direct references to functions).
+    """Metadata of a completed and registered `Pipeline` object.
+    Can be added to a Garden and execute on a remote Globus Compute endpoint.
 
-    Note:
-        Attributes are nearly identical to [Pipeline][garden_ai.pipelines.Pipeline], with a few exceptions.
+    `RegisteredPipelines` can be described completely by JSON
```

Comment:
I'm wondering if `RegisteredPipeline` should just become `PipelineMetadata`? Especially since it's going to serve a basically similar role to `ModelMetadata` in the new publishing workflow

Reply:
I agree, I think it should! I just think there was enough going on focusing on the deletions here without getting into renaming and refactoring. But I think I'll probably go for the rename in my next PR.